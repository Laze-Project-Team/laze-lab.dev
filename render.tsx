import type {
  Queries,
  queries,
  RenderOptions,
  RenderResult,
} from '@testing-library/react';
import { render as defaultRender } from '@testing-library/react';
import type { FC, ReactNode } from 'react';

import { ColorModeProvider } from '@/components/contexts/ColorModeContext';

export type renderType<
  Q extends Queries = typeof queries,
  Container extends Element | DocumentFragment = HTMLElement,
  BaseElement extends Element | DocumentFragment = Container,
> = (
  ui: React.ReactElement,
  options?: RenderOptions<Q, Container, BaseElement>,
) => RenderResult<Q, Container, BaseElement>;

type wrapperProps = {
  children?: ReactNode;
};

type globalWrapperHOCType = (
  wrapper?: React.JSXElementConstructor<{ children: React.ReactElement }>,
) => FC<wrapperProps>;

const globalWrapperHOC: globalWrapperHOCType = (Wrapper) => {
  const globalWrapper: FC<wrapperProps> = ({ children }) => (
    <ColorModeProvider>
      {Wrapper ? (
        <Wrapper>
          <>{children}</>
        </Wrapper>
      ) : (
        children
      )}
    </ColorModeProvider>
  );

  return globalWrapper;
};

export const render: renderType = (ui, options) =>
  defaultRender(ui, {
    ...options,
    wrapper: globalWrapperHOC(options?.wrapper),
  });
