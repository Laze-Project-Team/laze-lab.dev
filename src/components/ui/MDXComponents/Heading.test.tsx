import '@testing-library/jest-dom';

import { screen } from '@testing-library/react';
import { expect, it } from 'vitest';

import { render } from '@/lib/test/render';

import { H1, H2, H3, H4, H5, H6 } from '.';

it.each([
  ['H1', <H1 key="H1" />],
  ['H2', <H2 key="H2" />],
  ['H3', <H3 key="H3" />],
  ['H4', <H4 key="H4" />],
  ['H5', <H5 key="H5" />],
  ['H6', <H6 key="H6" />],
])('should be rendered correctly (%s)', (name, component) => {
  const { container } = render(component);
  expect(container.firstChild).toMatchSnapshot();
});

it.each([
  ['H1', <H1 key="H1" />],
  ['H2', <H2 key="H2" />],
  ['H3', <H3 key="H3" />],
  ['H4', <H4 key="H4" />],
  ['H5', <H5 key="H5" />],
  ['H6', <H6 key="H6" />],
])('should have "heading" role (%s)', (name, component) => {
  render(component);
  expect(screen.getByRole('heading')).toBeInTheDocument();
});
