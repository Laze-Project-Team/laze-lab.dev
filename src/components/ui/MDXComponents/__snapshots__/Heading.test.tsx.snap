// Vitest Snapshot v1

exports[`should be rendered correctly (H1) 1`] = `
<h1
  class="MuiTypography-root MuiTypography-h1 css-o2w69a-MuiTypography-root"
/>
`;

exports[`should be rendered correctly (H2) 1`] = `
<h2
  class="MuiTypography-root MuiTypography-h2 css-1sra7t5-MuiTypography-root"
/>
`;

exports[`should be rendered correctly (H3) 1`] = `
<h3
  class="MuiTypography-root MuiTypography-h3 css-gepadz-MuiTypography-root"
/>
`;

exports[`should be rendered correctly (H4) 1`] = `
<h4
  class="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
/>
`;

exports[`should be rendered correctly (H5) 1`] = `
<h5
  class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
/>
`;

exports[`should be rendered correctly (H6) 1`] = `
<h6
  class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
/>
`;
